cluster-autoscaler/estimator/threshold_based_limiter.go

```diff
@@ -29,17 +29,34 @@ type thresholdBasedEstimationLimiter struct {
 	maxNodes    int
 	nodes       int
 	start       time.Time
+	thresholds  []Threshold
 }
 
-func (tbel *thresholdBasedEstimationLimiter) StartEstimation([]*apiv1.Pod, cloudprovider.NodeGroup) {
+func (tbel *thresholdBasedEstimationLimiter) StartEstimation(_ []*apiv1.Pod, nodeGroup cloudprovider.NodeGroup, context EstimationContext) {
 	tbel.start = time.Now()
 	tbel.nodes = 0
+	tbel.maxNodes = 0
+	tbel.maxDuration = time.Duration(0)
+	for _, threshold := range tbel.thresholds {
+		tbel.maxNodes = getMinLimit(tbel.maxNodes, threshold.NodeLimit(nodeGroup, context))
+		tbel.maxDuration = getMinLimit(tbel.maxDuration, threshold.DurationLimit(nodeGroup, context))
+	}
+}
+
+func getMinLimit[V int | time.Duration](baseLimit V, targetLimit V) V {
+	if baseLimit < 0 || targetLimit < 0 {
+		return -1
+	}
+	if (baseLimit == 0 || baseLimit > targetLimit) && targetLimit > 0 {
+		return targetLimit
+	}
+	return baseLimit
 }
 
 func (*thresholdBasedEstimationLimiter) EndEstimation() {}
 
 func (tbel *thresholdBasedEstimationLimiter) PermissionToAddNode() bool {
-	if tbel.maxNodes > 0 && tbel.nodes >= tbel.maxNodes {
+	if tbel.maxNodes < 0 || (tbel.maxNodes > 0 && tbel.nodes >= tbel.maxNodes) {
```

Comment:
We should add the same check for duration below, to include negative durations.

Reply:
Corrected in b4213d82444eab039ba1fe0b6a0dbaf642deadc3